src/lib/yacc/grammar.rs

```diff
@@ -30,6 +30,7 @@
 // DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```

Comment:
I have to say I don't get this commit at all. Why do we need to introduce RefCells if we only want min or max costs. Could this not just be two different calls. If I call `max_costs` it calculates the maximal costs and if I call `min_costs` I get the min ones? Why does calculating one has anything to do with the other?

Reply:
I'm not quite sure which commit you're referring to, but I can probably guess.

They don't have anything to do with each other: the idea is that we lazily create the necessary min_costs/max_costs only when the user asks for something related to them. This means that one `SentenceGenerator` can handle both min/max sentences without being too expensive.

Reply:
Oh I thought I commented on the commit.

So before this change it wasn't lazily calculated?

Reply:
Correct.